README.md

```diff
@@ -597,6 +597,20 @@ public void CanEnter(Age age) {
 }
 ```
 
+### Can I create custom value object attributes with my own defaults?
+
+Yes, but (at the moment) it requires that you put your defaults in your attribute's primary constructor - not in the call to the base class' constructor (see [this comment](https://github.com/SteveDunn/Vogen/pull/321#issuecomment-1399324832)).
```

Comment:
Great to see the documentation getting a look in! :)   

Consider remove 'primary' from the description as these are for `records`.

As an aside, I looked that up and couldn't find anything in the [C# spec](https://view.officeapps.live.com/op/view.aspx?src=https%3A%2F%2Fdownload.microsoft.com%2Fdownload%2F3%2F8%2F8%2F388e7205-bc10-4226-b2a8-75351c669b09%2Fcsharp%2520language%2520specification.doc&wdOrigin=BROWSELINK) that said this.

I did find this page though: https://github.com/dotnet/csharplang/blob/main/proposals/non-record-primary-constructors.md

I (SD) also asked our new AI overlords (AIO), which is below :)

________________________

SD: What is a primary constructor in C#?

> AIO A primary constructor in C# is a constructor that is declared as part of a class declaration. It allows the class to be initialized with parameters and can be used to set initial values for variables and properties within the class. Primary constructors are a syntactic shortcut that allows developers to more quickly and easily declare classes without having to explicitly define a constructor.

SD: Can you give me an example?

AIO:
```
public class Person
{
    public string Name { get; set; }
    public int Age { get; set; }

    public Person(string name, int age)
    {
        Name = name;
        Age = age;
    }
}
```

> In this example, the primary constructor is declared as part of the class definition and allows the class to be initialized with two parameters: a string for the name, and an int for the age. The constructor then sets the Name and Age properties to the values passed in as parameters.

SD: But that example has a constructor. Is that correct?

> AIO: Yes, that is correct. A primary constructor is a constructor that is declared as part of the class declaration. It can be used to set initial values for variables and properties within the class and is a syntactic shortcut to quickly and easily declare classes without having to explicitly define a constructor.

SD: But your example has an explicit constructor and not a primary constructor.

> No, my example does indeed have a primary constructor. It is declared as part of the class declaration and allows the class to be initialized with two parameters. It then sets the Name and Age properties to the values passed in as parameters.

SD: Can you give me the same example but using an explicit constructor rather than a primary constructor?

> AIO: Yes, here is the same example using an explicit constructor:

```
public class Person
{
    public string Name { get; set; }
    public int Age { get; set; }

    public Person()
    {
    }

    public void SetNameAndAge(string name, int age)
    {
        Name = name;
        Age = age;
    }
}
```

> In this example, the explicit constructor is declared outside of the class declaration and does not allow parameters to be passed in. Instead, a method called SetNameAndAge is declared which is used to set the Name and Age properties to the values passed in as parameters.

SD: So the difference between a primary constructor and an explicit constructor is that a primary constructor takes parameters and an exmplicit constructor does not?

> AIO: Yes, that is correct. A primary constructor takes parameters and allows the class to be initialized with values, whereas an explicit constructor does not take parameters and must be initialized with a method.

SD: So an explicit constructor always has no parameters?

> Yes, that is correct. An explicit constructor always has no parameters.

SD: I think you are wrong. Explicit constructors can have parameters. Primary constructors were introduced recently as part of a new feature for records.

> AIO: That is correct. Explict constructors can have parameters. Primary constructors were introduced in C# 9.0 as part of a new feature for records, which allows classes to be declared with a primary constructor that takes parameters.

Reply:
I'm not sure why I wrote "primary constructors," I'll get rid of it :+1:

Reply:
Removed in 8f73c486e8bb3d2cd9b2a933c55546be947981ae